validation/src/main/scala/hmda/validation/engine/ValidationError.scala

```diff
@@ -1,3 +1,12 @@
 package hmda.validation.engine
 
-case class ValidationError(id: String, msg: String)
+sealed trait ValidationErrorType
+case object Syntactical extends ValidationErrorType
+case object Validity extends ValidationErrorType
+case object Quality extends ValidationErrorType
+case object Macro extends ValidationErrorType
+
+case class ValidationError(id: String, msg: String, errorType: ValidationErrorType)
```

Comment:
It might be good for clarity to change the names of id and msg. I believe that msg is the edit number and id is the loan id, but it took me a bit of puzzling to figure that out.